src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs

```diff
@@ -72,5 +80,288 @@ protected IList<TaggedText> GetAwaitableUsage(IMethodSymbol method, SemanticMode
 
             return SpecializedCollections.EmptyList<TaggedText>();
         }
+
+        /// <summary>
+        /// If the symbol could not be bound, we could be dealing with a partial invocation, we'll try to find a possible overload.
+        /// If it was bound, we'll find which parameter to highlight.
+        /// Either way, we'll eliminate unsuitable candidates (based on argument names used in the invocation).
+        /// </summary>
+        protected static (ISymbol symbol, int parameterIndex) GuessCurrentSymbolAndParameter(
+            SeparatedSyntaxList<ArgumentSyntax> arguments, ImmutableArray<IMethodSymbol> methodGroup, int position,
+            SemanticModel semanticModel, ISemanticFactsService semanticFactsService, CancellationToken cancellationToken)
+        {
+            if (arguments.Count != 0)
+            {
+                foreach (var method in methodGroup)
+                {
+                    if (IsAcceptable(arguments, method, position, semanticModel, semanticFactsService, out var parameterIndex))
+                    {
+                        return (method, parameterIndex);
+                    }
+                }
+            }
+
+            // Note: Providing no recommendation if no arguments allows the model to keep the last implicit choice
+            return (null, -1);
+        }
+
+        /// <summary>
+        /// If an argument name does not correspond to a parameter name, this method is unacceptable. We'll filter it out.
+        /// </summary>
+        protected static bool IsUnacceptable(SeparatedSyntaxList<ArgumentSyntax> arguments, IMethodSymbol method)
+        {
+            if (arguments.Count == 0)
+            {
+                return false;
+            }
+
+            var parameterNames = method.Parameters.SelectAsArray(p => p.Name);
+            foreach (var argument in arguments)
+            {
+                if (HasName(argument, out var name))
```

Comment:
if you end up moving this to a common location for VB and C# then you can use ISyntaxFactsService.GetNameForArgument

Reply:
still applies.